server/src/main/java/org/opensearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -822,8 +823,9 @@ public static Translog.Location performOnReplica(BulkShardRequest request, Index
             }
             assert operationResult != null : "operation result must never be null when primary response has no failure";
             location = syncOperationResultOrThrow(operationResult, location);
+            maxSeqNo = response.getResponse().getSeqNo();
```

Comment:
we need to keep track of ongoing max here as we iterate all the items, this will overwrite the value with each iteration.  The last item in the list is not guaranteed to have the highest seqNo.

Reply:
makes sense. I will keep track of maxSeqNo here